### PR TITLE
Added code for Conversion of Binary to Decimal

### LINKED ISSUE #3060: Conversion of Binary Number to Decimal Number in C++
## 💥 Proposal

(A clear and concise description of what the proposal is.)

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Hey I am GSSOC'21 Participant and I have read the Contributing Guidelines.
I want to propose a program in which Binary Number (up to 10 digits) will be given as input and in output we get the number in its Decimal form.
Example: Input 1000 Output 8
Please assign me this issue.

### PR DESCRIPTION
**Pull request by sanskruti-shahu**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes, I have read the guidelines.

### Description

Added code for Conversion of Binary  to Decimal

### Checklist

- [ Yes ] I've read the contribution guidelines.
- [ Yes ] I've checked the issue list before deciding what to submit.
- [ Yes ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #3060

### CODE REVIEW

**Comment by sanskruti-shahu** [comment]
@ankitaggarwal23 Please review this PR.

**Comment by sanskruti-shahu** [comment]
@HarshCasper Please review my PR

File: C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/* C++ Program for Conversion of Binary Number to Decimal Number
+In this Program only input line contains an integer input 'number'. Its length does not exceed 10.
+Here, a function named binary_to_decimal will convert the given binary input into decimal number.
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp]: Keep `binary_to_decimal` in backticks

File: C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+And output contain Decimal Conversion of input in a single line.
+*/  
+#include<bits/stdc++.h>
+using namespace std;
+int binary_to_decimal(int number){
+    int i, remainder, store, output;
+    //initialising variables
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp]: Redundant Comments

File: C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+*/  
+#include<bits/stdc++.h>
+using namespace std;
+int binary_to_decimal(int number){
+    int i, remainder, store, output;
+    //initialising variables     
+    remainder=1;
+    store=1;
+    output=0;
+    //loop to perform conversion task
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp]: Redundant Comments

File: C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+using namespace std;
+int binary_to_decimal(int number){
+    int i, remainder, store, output;
+    //initialising variables     
+    remainder=1;
+    store=1;
+    output=0;
+    //loop to perform conversion task    
+    for(i=0;number>0;i++){
+        //dividing number by 10 to get its digits of one's place, ten's place, etc.. in each iteration respectively
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp]: Make it shorter

File: C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    int i, remainder, store, output;
+    //initialising variables     
+    remainder=1;
+    store=1;
+    output=0;
+    //loop to perform conversion task    
+    for(i=0;number>0;i++){
+        //dividing number by 10 to get its digits of one's place, ten's place, etc.. in each iteration respectively    
+        remainder=number%10;
+        //multiplying remainder with power of 2 according to its place value
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp:16]: Make it shorter

File: C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        store=remainder*(pow(2,i));
+        //storing this in each iteration to get the final output at end
+        output=output+store;
+        //modifying number
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp]: Redundant Comment

File: C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        number=number/10;
+    }
+    return output;
+}
+int main(){
+    int numberInput,l,k;
+    //getting user input
+    cin>>numberInput;
+    //calling function to perform the task
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp]: Redundant Comment

File: C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        output=output+store;
+        //modifying number
+        number=number/10;
+    }
+    return output;
+}
+int main(){
+    int numberInput,l,k;
+    //getting user input
+    cin>>numberInput;
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp:26]: Redundant Comment

File: C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+OUTPUT  4
+Testcase 2
+INPUT   1111
+OUTPUT  15
+Testcase 3
+INPUT   111000
+OUTPUT  56   
+Time Complexity is O(t)    where t is number of digit in 'number'
+Space Complexity is O(1)   */
+
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp]: Remove the extra line of Code

File: C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+Testcase 1
+INPUT   100
+OUTPUT  4
+Testcase 2
+INPUT   1111
+OUTPUT  15
+Testcase 3
+INPUT   111000
+OUTPUT  56   
+Time Complexity is O(t)    where t is number of digit in 'number'
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp]: Keep `t` in capital

File: C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+/* SAMPLE
+Testcase 1
+INPUT   100
+OUTPUT  4
+Testcase 2
+INPUT   1111
+OUTPUT  15
+Testcase 3
+INPUT   111000
+OUTPUT  56
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Conversion_from_Binary_to_Decimal.cpp]: Remove extra test-cases

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Format the Code as well

**Comment by sanskruti-shahu** [comment]
@ankitaggarwal23 @HarshCasper I updated changes please review

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**